wcs-functions.php

```diff
@@ -134,77 +139,53 @@ function wcs_create_subscription( $args = array() ) {
 	}
 
-	// check customer id is set
+	// Check customer id is set.
 	if ( empty( $args['customer_id'] ) || ! is_numeric( $args['customer_id'] ) || $args['customer_id'] <= 0 ) {
 		return new WP_Error( 'woocommerce_subscription_invalid_customer_id', _x( 'Invalid subscription customer_id.', 'Error message while creating a subscription', 'woocommerce-subscriptions' ) );
 	}
 
-	// check the billing period
+	// Check the billing period.
 	if ( empty( $args['billing_period'] ) || ! in_array( strtolower( $args['billing_period'] ), array_keys( wcs_get_subscription_period_strings() ) ) ) {
 		return new WP_Error( 'woocommerce_subscription_invalid_billing_period', __( 'Invalid subscription billing period given.', 'woocommerce-subscriptions' ) );
 	}
 
-	// check the billing interval
+	// Check the billing interval.
 	if ( empty( $args['billing_interval'] ) || ! is_numeric( $args['billing_interval'] ) || absint( $args['billing_interval'] ) <= 0 ) {
 		return new WP_Error( 'woocommerce_subscription_invalid_billing_interval', __( 'Invalid subscription billing interval given. Must be an integer greater than 0.', 'woocommerce-subscriptions' ) );
 	}
 
-	$subscription_data['post_type']     = 'shop_subscription';
-	$subscription_data['post_status']   = 'wc-' . apply_filters( 'woocommerce_default_subscription_status', 'pending' );
-	$subscription_data['ping_status']   = 'closed';
-	$subscription_data['post_author']   = 1;
-	$subscription_data['post_password'] = uniqid( 'order_' );
-	// translators: Order date parsed by strftime
-	$post_title_date = strftime( _x( '%b %d, %Y @ %I:%M %p', 'Used in subscription post title. "Subscription renewal order - <this>"', 'woocommerce-subscriptions' ) ); // phpcs:ignore WordPress.WP.I18n.UnorderedPlaceholdersText
-	// translators: placeholder is order date parsed by strftime
-	$subscription_data['post_title']    = sprintf( _x( 'Subscription &ndash; %s', 'The post title for the new subscription', 'woocommerce-subscriptions' ), $post_title_date );
-	$subscription_data['post_date_gmt'] = $args['date_created'];
-	$subscription_data['post_date']     = get_date_from_gmt( $args['date_created'] );
-
-	if ( $args['order_id'] > 0 ) {
-		$subscription_data['post_parent'] = absint( $args['order_id'] );
-	}
-
-	if ( ! is_null( $args['customer_note'] ) && ! empty( $args['customer_note'] ) ) {
-		$subscription_data['post_excerpt'] = $args['customer_note'];
-	}
+	$subscription = new \WC_Subscription();
 
-	// Only set the status if creating a new subscription, use wcs_update_subscription to update the status
 	if ( $args['status'] ) {
-		if ( ! in_array( 'wc-' . $args['status'], array_keys( wcs_get_subscription_statuses() ) ) ) {
-			return new WP_Error( 'woocommerce_invalid_subscription_status', __( 'Invalid subscription status given.', 'woocommerce-subscriptions' ) );
-		}
-		$subscription_data['post_status']  = 'wc-' . $args['status'];
+		$subscription->set_status( $args['status'] );
```

Comment:
Suggest keeping the comment unless it's irrelevant (it's helping me understand what's happening here 😝)
`// Only set the status if creating a new subscription, use wcs_update_subscription to update the status`

Reply:
Fixed in ae559089